demo/app.js

```diff
@@ -1,6 +1,143 @@
-/*global document:false*/
+/*global document:false window:false */
+/*eslint-disable react/no-multi-comp */
 import React from "react";
-import { render } from "react-dom";
-import Demo from "./demo";
+import ReactDOM from "react-dom";
+
+import AreaDemo from "./components/victory-area-demo";
+import AxisDemo from "./components/victory-axis-demo";
+import BarDemo from "./components/victory-bar-demo";
+import ChartDemo from "./components/victory-chart-demo";
+import LineDemo from "./components/victory-line-demo";
+import ScatterDemo from "./components/victory-scatter-demo";
+import ErrorBarDemo from "./components/victory-errorbar-demo";
+import BoxplotDemo from "./components/victory-boxplot-demo";
+import CandlestickDemo from "./components/victory-candlestick-demo";
+import EventsDemo from "./components/events-demo";
+import GroupDemo from "./components/group-demo";
+import VoronoiDemo from "./components/victory-voronoi-demo";
+import TooltipDemo from "./components/victory-tooltip-demo";
+import ZoomContainerDemo from "./components/victory-zoom-container-demo";
+import VoronoiContainerDemo from "./components/victory-voronoi-container-demo";
+import CursorContainerDemo from "./components/victory-cursor-container-demo";
+import CreateContainerDemo from "./components/create-container-demo";
+import BrushContainerDemo from "./components/victory-brush-container-demo";
+import AnimationDemo from "./components/animation-demo";
+import SelectionDemo from "./components/selection-demo";
+import PolarDemo from "./components/victory-polar-chart-demo";
+import ImmutableDemo from "./components/immutable-demo";
+import ExternalEventsDemo from "./components/external-events-demo";
+import VictoryBrushLineDemo from "./components/victory-brush-line-demo";
+import PerformanceDemo from "./components/performance";
+import DebugDemo from "./components/debug-demo";
+import VictoryLabelDemo from "./components/victory-label-demo";
+import VictoryLegendDemo from "./components/victory-legend-demo";
+import VictoryPieDemo from "./components/victory-pie-demo";
+import VictoryDemo from "./components/victory-demo";
+
+class Home extends React.Component {
+  render() {
+    return (
+      <h1>Pick A Demo</h1>
+    );
+  }
+}
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      route: window.location.hash.substr(1)
+    };
+  }
+
+  componentWillMount() {
+    window.addEventListener("hashchange", () => {
+      this.setState({
+        route: window.location.hash.substr(1)
+      });
+    });
+  }
+
+  getDemo() { // eslint-disable-line complexity
+    let Child;
```

Comment:
NIT: I would have done an object map instead like:

```js
const MAP = {
  "/axis": AxisDemo,
  // ...
};

getDemo() {
  return MAP[this.state.route] || Home;
}
```

to avoid the monster `switch` statement.